src/main/sensors/voltage.c

```diff
@@ -163,17 +171,26 @@ void voltageMeterADCRefresh(void)
 
         uint8_t channel = voltageMeterAdcChannelMap[i];
         uint16_t rawSample = adcGetChannel(channel);
-
-        uint16_t filteredSample = pt1FilterApply(&state->filter, rawSample);
+        uint16_t filteredDisplaySample = pt1FilterApply(&state->displayFilter, rawSample);
 
         // always calculate the latest voltage, see getLatestVoltage() which does the calculation on demand.
-        state->voltageFiltered = voltageAdcToVoltage(filteredSample, config);
+        state->voltageDisplayFiltered = voltageAdcToVoltage(filteredDisplaySample, config);
         state->voltageUnfiltered = voltageAdcToVoltage(rawSample, config);
+
+#if defined(USE_BATTERY_VOLTAGE_SAG_COMPENSATION)
+        if (currentPidProfile->vbat_sag_compensation > 0) {
+            uint16_t filteredSagSample = pt1FilterApply(&state->sagFilter, rawSample);
+            state->voltageSagFiltered = voltageAdcToVoltage(filteredSagSample, config);
+        }
```

Comment:
Unless we're going to get more complicated with the profile switching and filter initialization, then we need to continue to apply the samples to the filter if **any profile** has the feature enabled. Otherwise we'd have a discontinuity in the filter state if the user switched from a profile that had it disabled to one that has it enabled. See also the comment that the filter needs to be initialized **if any** profile has the feature.